### ISSUE TITLE
Subprocess works in IPython but not in Notebook

### ISSUE THREAD

**Issue by connorferster** [jupyter/notebook#5252]
I am a bit perplexed by this issue and do not know where to start to tackle it. 

I am writing a script to call:
```python
subprocess.run(['pdflatex', 'MyFileName.tex'])
```
And it works in IPython but not within a Notebook. I also did some further tests:
```python
subprocess.run(['ls -l', shell = True])
subprocess.run(["ls", "-l"])
```
These work in IPython but not in a Notebook (error code 2).

```python
subprocess.run(['who'])
```
This, however, works in both IPython and Jupyter. My hunch is that commands that I can send that do not require arguments seem to work where as commands that need arguments seem not to. 

I am running Jupyter 1.0 on Ubuntu 18.04LTS with Python 3.7.4

I saw issue #1018 but it does not appear to be quite the same issue.

Any thoughts on this? Thank you so much for any time you can allocate to this.

**Comment by kevin-bates** [comment]
Hmm - I'm not reproducing this.  Here's what I see.  (I had to move the right bracket on the 'shell=True' command.)  I also added the "bang" format (`!ls -l`) in case that's something to try as a data point.
![image](https://user-images.githubusercontent.com/22599560/75381205-34577000-588d-11ea-82e3-fe46b04bd6c8.png)

Given your results, this is sounding more like an environmental issue, although we should probably confirm your version details since "1.0" is not correct relative to Notebook or its required projects.

What does the output of `jupyter --version` produce?

It might also be worth checking your PATH via `!echo $PATH`.  Perhaps the env instantiated on behalf of the kernel is missing some fundamental directory (e.g., /bin) that causes basic commands like 'ls' to appear as though they're not available, yielding `ENOENT`.

**Comment by connorferster** [comment]
Thank you for your response. Agreed: I don't know where I got "1.0" from... Here is the output from `jupyter --version`:
```
jupyter core     : 4.5.0
jupyter-notebook : 6.0.1
qtconsole        : 4.5.5
ipython          : 7.8.0
ipykernel        : 5.1.2
jupyter client   : 5.3.3
jupyter lab      : 1.1.4
nbconvert        : 5.6.0
ipywidgets       : 7.5.1
nbformat         : 4.4.0
traitlets        : 4.3.3
```

Interestingly, bang format works every time! All the commands worked from Jupyter. I also used it to check the PATH variable, as you suggested. Everything looked normal there (as far as I can see, anyway): 

```
/home/juno/.pyenv/versions/anaconda3-2019.10/bin:/home/juno/.pyenv/libexec:/home/juno/.pyenv/plugins/python-build/bin:/home/juno/.local/bin:/home/juno/.pyenv/shims:~/.local/bin:/home/juno/.pyenv/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
```

I am assuming the return code of `1` in your `pdflatex` cell is because you did not actually have a `"MyFileName.tex"` but that is the return code that I am also getting from my script, which is what I was originally trying to fix. It seems that a latex compile error occurs (so says my log) when run from within Jupyter but when run from the command line or from IPython, the latex compiler has no problem at the compiling point that it previously had an issue with.

I also don't have this issue on my Windows machine.

Let me know if you have any other thoughts based on this information. I am grateful for your help thus far. If you nothing comes to mind, I will keep playing with this and post something when I hopefully get it solved.

**Comment by kevin-bates** [comment]
Correct, the .tex file didn't exist.  I downloaded a sample.tex file and still received return code 1, but that was also produced from the shell.  Downloading a different sample, I was able to create a pdf file from both the shell and notebook (returncode 0).

Your PATH seems fine - no glaring ommissions of basic directories.  I'm assuming things like `ls` are in `/bin` or `/usr/bin` in your installation.

Very strange.  I'm not thinking of anything at the moment.  I think the area of focus would be ipykernel and ipython.  Since ipython appears to work directly, that would likely leave the former.  Can you confirm you're using the same ipykernel version on Ubuntu and Windows?

I'll try to keep this in my "retention buffer" in case anything comes to mind.  Thanks for coming back to update the issue should you find anything.
